src/Tasks/AssemblyDependency/AssemblyInformation.cs

```diff
@@ -866,19 +866,30 @@ private class HeaderInfo
         }
 
         /// <summary>
-        /// Given a path get the CLR runtime version of the file
+        /// Given a path get the CLR runtime version of the file.
         /// </summary>
         /// <param name="path">path to the file</param>
         /// <returns>The CLR runtime version or empty if the path does not exist or the file is not an assembly.</returns>
         public static string GetRuntimeVersion(string path)
         {
-            using (var sr = new BinaryReader(File.OpenRead(path)))
+            if (!FileSystems.Default.FileExists(path))
             {
-                if (!FileSystems.Default.FileExists(path))
-                {
-                    return string.Empty;
-                }
+                return string.Empty;
+            }
 
+            Stream stream = File.OpenRead(path);
+            return GetRuntimeVersion(stream);
+        }
+
+        /// <summary>
+        /// Given a stream get the CLR runtime version of the underlying file.
+        /// </summary>
+        /// <param name="stream">A stream representing the file</param>
+        /// <returns>The CLR runtime version or empty if the stream does not represent an assembly.</returns>
+        internal static string GetRuntimeVersion(Stream stream)
+        {
+            using (var sr = new BinaryReader(stream))
```

Comment:
Interesting choice that calee is disposing the resource instead of the owner (caller).
But since it's internal and single usage method - it probably  doesn't matter

Reply:
You're bringing up a great point. Does disposing of the `BinaryReader` close the file on this code path? I have kept the calls exactly like before but I wonder if this is correct.

Reply:
Aha, that's one of the weirder BCL design choices. `BinaryReader` closes the underlying stream by default.
https://learn.microsoft.com/en-us/dotnet/api/system.io.binaryreader.-ctor?view=net-7.0#system-io-binaryreader-ctor(system-io-stream-system-text-encoding-system-boolean)

Though not needed, I'll add a `Dispose` call to the caller for clarity. Thank you!

Reply:
Yeah - decoration and disposing is always bit tricky.
IIRC the decorating streams can usualy be instructed to not dispose the underlying streams.

The other option is not to disect the boundary between the caller-calee over the decorating chain, but rather let just one of those handle all streams. Again - overkill adjustments for this one-off scenario

Reply:
Done!